src/fpm_meta.f90

```diff
@@ -1598,4 +1600,124 @@ type(string_t) function mpi_wrapper_query(mpilib,wrapper,command,verbose,error)
 
 end function mpi_wrapper_query
 
+!> Check if input is a useful linker argument
+logical function is_link_argument(compiler,string)
+   type(compiler_t), intent(in) :: compiler
+   character(*), intent(in) :: string
+
+   select case (compiler%id)
+      case (id_intel_classic_windows,id_intel_llvm_windows)
+          is_link_argument = string=='/link' &
+                             .or. str_begins_with_str(string,'/LIBPATH')&
+                             .or. str_ends_with(string,'.lib') ! always .lib whether static or dynamic
+      case default
+
+          ! fix OpenMPI's Fortran wrapper bug (https://github.com/open-mpi/ompi/issues/11636) here
+          is_link_argument = (    str_begins_with_str(string,'-L') &
```

Comment:
Should this logic be invoked only for OpenMPI wrappers in a certain version range? Or would is the extra effort unwarranted?

Reply:
Here, the [meson](https://github.com/mesonbuild/meson/blob/ebf5757c59d9bceb953dc531c757e07dcf83c081/mesonbuild/dependencies/mpi.py#L134C1-L135C1) approach is replicated, i.e., to remove all flags that are not strictly 1) library directories 2) library flags 3) important warning flags. It probably shielded it from the macOS issue above, and may as well do the same in the future